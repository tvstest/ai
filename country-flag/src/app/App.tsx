import React from 'react'
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom'
import CountrySearch from 'app/pages/CountrySearch'
import CountryList from 'app/pages/CountryList'
import Navbar from 'app/components/Navbar'
import { Routes } from 'app/utils/enums/routes'
import { ThemeProvider } from '@mui/material'
import appTheme from 'app/configs/theme'

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={appTheme}>
        <Navbar />
        <Switch>
          <Route exact path={Routes.Home} component={CountrySearch} />
          <Route exact path={Routes.CountriesByName} component={CountryList} />
          <Redirect to={Routes.Home} />
        </Switch>
      </ThemeProvider>
    </BrowserRouter>
  )
}
export default App
