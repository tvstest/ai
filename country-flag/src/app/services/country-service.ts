import { ICountry } from "app/utils/interfaces/country";
import { AxiosResponse } from "axios";
import { countryHttpClient } from "app/services/http.service";

const getAll = (): Promise<AxiosResponse<ICountry[]>> =>
  countryHttpClient.get(`/all`);

const getByName = (countryName: string): Promise<AxiosResponse<ICountry[]>> =>
  countryHttpClient.get(`/name/${countryName}`);

export const countryService = {
  getByName,
  getAll,
};
export default countryService;
