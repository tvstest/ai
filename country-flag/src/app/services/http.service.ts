import { COUNTRIES_API_BASE_URL, WEATHER_API_BASE_URL } from "app/utils/constants";
import axios from "axios";

const httpClient = axios;
export const countryHttpClient = httpClient.create({
  baseURL: COUNTRIES_API_BASE_URL,
});
export const weatherHttpClient = httpClient.create({
  baseURL: WEATHER_API_BASE_URL,
});
