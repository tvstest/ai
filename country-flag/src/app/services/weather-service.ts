import { AxiosResponse } from "axios";
import { weatherHttpClient } from "app/services/http.service";
import { ICapitalWeatherInfo } from "app/utils/interfaces/weather";

const getWeatherByCity = (
  cityName: string
): Promise<AxiosResponse<ICapitalWeatherInfo>> =>
  weatherHttpClient.get(
    `/current?access_key=${process.env.REACT_APP_WEATHER_API_KEY}&query=${cityName}`
  );

export const weatherService = {
  getWeatherByCity,
};

export default weatherService;
