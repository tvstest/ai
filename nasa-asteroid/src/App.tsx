import { ThemeProvider } from '@mui/material/styles'
import Navbar from 'app/components/Navbar'
import Routes from 'app/components/Routes'
import { Suspense } from 'react'
import theme from './app/config/theme'

const App: React.FC = () => {
  return (
    <Suspense fallback="Loading ...">
      <Navbar />
      <ThemeProvider theme={theme}>
        <Routes />
      </ThemeProvider>
    </Suspense>
  )
}

export default App
