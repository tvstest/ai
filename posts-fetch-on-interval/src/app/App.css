.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* CSS Loader
--------------------------------------------------------------------------------------------------------*/
.loader-overlay {
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  background: rgba(215, 215, 215, 0.35);
  z-index: 9999;
}
.cssload-fond {
  margin: auto;
  position: fixed;
  z-index: 0.5;
  top: 50%;
  left: 50%;
  margin: -20px 0 0 -20px;
}

.loaderHide {
  display: none;
}

.loaderShow {
  display: block;
}
