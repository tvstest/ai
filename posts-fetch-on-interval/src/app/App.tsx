import React, { Suspense } from 'react'
import { ThemeProvider } from '@mui/material/styles'
import { ToastContainer } from 'react-toastify'
import Loader from './components/Loader'
import Navbar from './components/Navbar'
import { theme } from './app.theme'
import 'react-toastify/dist/ReactToastify.css'
import './App.css'
import PostsData from './components/PostsData'

function App() {
  return (
    <Suspense fallback={<Loader />}>
      <ThemeProvider theme={theme}>
        <ToastContainer />
        <Navbar />
        <PostsData />
        <Loader />
      </ThemeProvider>
    </Suspense>
  )
}

export default App
