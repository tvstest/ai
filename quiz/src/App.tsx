import './App.css'
import { CircularProgress, ThemeProvider } from '@mui/material'
import { Suspense } from 'react'
import Routes from 'components/Routes'
import theme from 'configs/app-theme'
import { BrowserRouter as Router } from 'react-router-dom'

const App: React.FC = () => {
  return (
    <Suspense fallback={<CircularProgress />}>
      <ThemeProvider theme={theme}>
        <Router>
          <Routes />
        </Router>
      </ThemeProvider>
    </Suspense>
  )
}

export default App
