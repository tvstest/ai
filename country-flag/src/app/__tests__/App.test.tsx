import App from 'app/App'
import { act, fireEvent, render, screen, waitFor } from '@testing-library/react'
import { countryService } from 'app/services/country-service'
import renderer from 'react-test-renderer'

beforeEach(() => {
  render(<App />)
})

test('matches the app snapshot', () => {
  const domTree = renderer.create(<App />).toJSON()
  expect(domTree).toMatchSnapshot()
})

test('get India country info and its capital weather info for Delhi', () => {
  const headerText = screen.getByText(/Country and Weather Info/i)
  expect(headerText).toBeInTheDocument()
})

test('specific country is found via api call', async () => {
  const countryServiceMock = jest.spyOn(countryService, 'getByName')
  const inputElement = screen.getByTestId('country-search-input')

  await act(async () => {
    fireEvent.change(inputElement, {
      target: { value: 'india' },
    })
  })
  await act(async () => {
    const buttonElement = screen.getByTestId('country-search-button')
    fireEvent.click(buttonElement)
  })

  expect(countryServiceMock).toHaveBeenLastCalledWith('india')

  await waitFor(
    async () => {
      const contentText = screen.getByText('India')
      expect(contentText).toBeInTheDocument()
    },
    { timeout: 5000 }
  )
})
