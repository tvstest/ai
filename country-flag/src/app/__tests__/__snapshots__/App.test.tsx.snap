// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the app snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular makeStyles-toolbar-1 css-1t29gy6-MuiToolbar-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 makeStyles-h5-2 css-n4yuy5-MuiTypography-root"
      >
        Country and Weather Info
      </h5>
    </div>
  </header>,
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <div
      className="makeStyles-paper-3"
    >
      <form
        autoComplete="off"
        className="makeStyles-form-5"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root makeStyles-fields-4 css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pw3wan-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            Enter Country
            <span
              aria-hidden={true}
              className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-sn8fib-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={true}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="country-search-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-173wfxe"
              >
                <span>
                  Enter Country
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled makeStyles-submit-6 css-1rcmftp-MuiButtonBase-root-MuiButton-root"
          data-testid="country-search-button"
          disabled={true}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </main>,
]
`;
